src/RegisterDevicePage.tsx

```diff
@@ -383,7 +383,11 @@ export default (): React.ReactElement => {
             <Button
               variant="primary"
               type="submit"
-              disabled={!isValidDeviceId || isRegisteringDevice}
+              disabled={
+                !isValidDeviceId ||
+                isRegisteringDevice ||
+                !astarte.token?.can('pairing', 'POST', '/devices')
+              }
```

Comment:
It's better to directly hide the button instead of displaying it disabled.

In addition, I would check this permission instead:
```tsx
astarte.token?.can('pairing', 'POST', '/agent/devices')
```